controllers/useraccount/console_usersettings_test.go

```diff
@@ -0,0 +1,285 @@
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-settings-name-no-match",
+					Namespace: UserSettingNS,
+					Labels: map[string]string{
+						ConsoleUserSettingsIdentifier: "true",
+						ConsoleUserSettingsUID:        "johnsmith",
+					},
+				},
+			}
+			cl := fake.NewClientBuilder().WithObjects(cm).Build()
+			err := deleteResource(context.TODO(), cl, "johnsmith", cm)
+			require.NoError(t, err)
+			// check that the configmap doesn't exist anymore
+			err = cl.Get(context.TODO(), client.ObjectKey{Name: "user-settings-name-no-match", Namespace: UserSettingNS}, &corev1.ConfigMap{})
+			require.True(t, errors.IsNotFound(err))
+		})
+		t.Run("multiple configmaps found by label and deletes successfully", func(t *testing.T) {
+			cm1 := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-settings-name-no-match",
+					Namespace: UserSettingNS,
+					Labels: map[string]string{
+						ConsoleUserSettingsIdentifier: "true",
+						ConsoleUserSettingsUID:        "johnsmith",
+					},
+				},
+			}
+			cm2 := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-settings-name-no-match-second",
+					Namespace: UserSettingNS,
+					Labels: map[string]string{
+						ConsoleUserSettingsIdentifier: "true",
+						ConsoleUserSettingsUID:        "johnsmith",
+					},
+				},
+			}
+			cl := fake.NewClientBuilder().WithObjects(cm1, cm2).Build()
+			err := deleteResource(context.TODO(), cl, "johnsmith", &corev1.ConfigMap{})
+			require.NoError(t, err)
+			// check that the configmaps don't exist anymore
+			err = cl.Get(context.TODO(), client.ObjectKey{Name: "user-settings-name-no-match", Namespace: UserSettingNS}, &corev1.ConfigMap{})
+			require.True(t, errors.IsNotFound(err))
+			err = cl.Get(context.TODO(), client.ObjectKey{Name: "user-settings-name-no-match-second", Namespace: UserSettingNS}, &corev1.ConfigMap{})
+			require.True(t, errors.IsNotFound(err))
+		})
+
+		t.Run("Error is returned when error in deleting configmap", func(t *testing.T) {
+			cl := test.NewFakeClient(t)
+			cl.MockGet = func(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+				return nil
+			}
+			cl.MockDelete = func(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error {
+				return fmt.Errorf("error in deleting configmap")
+			}
+			err := deleteResource(context.TODO(), cl, "johnsmith", &corev1.ConfigMap{})
+			require.Error(t, err)
+			require.Equal(t, "error in deleting configmap", err.Error())
+		})
+		t.Run("No Error is returned when no configmap not found", func(t *testing.T) {
+			cl := test.NewFakeClient(t)
+			err := deleteResource(context.TODO(), cl, "johnsmith", &corev1.ConfigMap{})
+			require.NoError(t, err)
+		})
+
+	})
+
+	t.Run("Delete Role", func(t *testing.T) {
```

Comment:
the `deleteResource` is generic now - it technically works with any type of the client.Object, so it doesn't matter if you test it with a ConfigMap, Role, or RoleBinding.
Let's pick one type, the rest looks like a duplication of the same. 
As it is written now, it's the responsibility of the caller (UserAccount controller) that should call the function with all three kinds/types.

Reply:
done, removed test cases for Role and rolebinding and modified the remaining test case description with configmap to indiacte object